### ISSUE TITLE
Can I change class_weight during training in Keras?

### ISSUE THREAD

**Issue by jeongHwarr** [keras-team/keras#12554]
I want to change my class_weight during training in Keras.

I used `fit_generator` and `Callback` method like below.

```
model.fit_generator(
                decoder_generator(x_train, y_train),
                steps_per_epoch=len(x_train),
                epochs=args.epochs,
                validation_data=decoder_generator(x_valid, y_valid),
                validation_steps=len(x_valid),
                callbacks=callback_list,
                class_weight=class_weights,
                verbose=1)
```
and

```
class Valid_checker(keras.callbacks.Callback):
    def __init__(self, model_name, patience, val_data, x_length):
        super().__init__()
        self.best_score = 0
        self.patience = patience
        self.current_patience = 0 
        self.model_name = model_name
        self.validation_data = val_data
        self.x_length = x_length


    def on_epoch_end(self, epoch, logs={}):
        X_val, y_val = self.validation_data
        y_predict, x_predict = model.predict_generator(no_decoder_generator(X_val, y_val), steps=len(X_val))
        y_predict = np.asarray(y_predict)
        x_predict = np.asarray(x_predict)  
```

`decoder_generator` and `no_decoder_generator` are just custom generator.

I want to change the class weight every time the epoch ends. Is it possible? then how can I do?

My data is imbalanced data, and overfitting is continued for one class.

At the end of the epoch I want to increase the weight for classes with low accuracy by calculating the accuracy by class.

How can I do?

**Comment by jannctu** [comment]
I don't think it's possible using fit_generator, you can try using train_on_batch instead.  
another possibility custom the loss function and do class balancing at there.
